### ISSUE TITLE
Quick fix for KeyError: 'dataset'

### ISSUE THREAD

**Issue by jtydhr88** [jexom/sd-webui-depth-lib#36]
Hi there,
If you are facing KeyError: 'dataset' issue, you can fix by:
1. go to your `webui_folder\venv\Lib\site-packages\gradio_client` and edit `serializing.py` file
2. go to the end of the serializing.py, and add one line inside COMPONENT_MAPPING:
`"dataset": StringSerializable,`
it should look like after modifing:
![image](https://github.com/jexom/sd-webui-depth-lib/assets/860985/8d0027b4-1f1f-4312-9061-efecc7e65dfb)
3. Save
4. Restart your webui.

it should work well now, I tested 900 hands on webui latest version 1.5.1 + gradio 3.32.0 + gradio_client 0.3.0

it should fix the issues:
1. https://github.com/jexom/sd-webui-depth-lib/issues/32 
2. https://github.com/jexom/sd-webui-depth-lib/issues/35
3. https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/11855

**Comment by jtydhr88** [comment]
I know webui plans to update gradio to latest(3.39) in later release v1.6.0 (see https://github.com/richrobber2/canvas-zoom/issues/88), but likely it still lost dataset map in https://github.com/gradio-app/gradio/blob/v3.39.0/client/python/gradio_client/serializing.py

**Comment by jtydhr88** [comment]
there are several ways to fix the issue permanently:
1. I am not sure the reason why gradio miss this code, it may have some reasons, but it could be gradio client bug, should be fixed in gradio side.
2. replace the code https://github.com/jexom/sd-webui-depth-lib/blob/efa9f616b30ea0f27e5dadf40ab41815012d1d78/scripts/main.py#L47 using gr.Examples by another way to show the image lib, may could be gallery component https://www.gradio.app/docs/gallery I guess.
3. totally give up gradio component and use other tech, like pure javascript instead to display the image lib (as a webui extension developer, I prefer to rebuild my page by HTML/CSS/JS rather than using gradio honstlly)

**Comment by alexbofa** [comment]
Hello. Thank you, helped with the replacement `serializing.py`

Do you know why don't show pictures? There are no errors in the console. I think it's because of the symbolic link to the extension https://github.com/jexom/sd-webui-depth-lib/issues/29
![Screen-1544](https://github.com/jexom/sd-webui-depth-lib/assets/58225118/5c1b4fce-318c-4d91-943e-db8dbd36d67d)

**Comment by jtydhr88** [comment]
@alexbofa i don’t see the issue that images not show up on my side, I tested the default images from this extension and 900 hands, and they work well. If you could provide more information I may help.

**Comment by webcoderz** [comment]
works for me also

**Comment by webcoderz** [comment]
gr.gallery solution looks okay and possible

**Comment by alexbofa** [comment]
> @alexbofa i don’t see the issue that images not show up on my side, I tested the default images from this extension and 900 hands, and they work well. If you could provide more information I may help.

I didn't have time to fix the message, because I found a problem in the issue. It's possible, as I wrote in the updated post, that the problem is that I'm using a symbolic link for "Extensions" https://github.com/jexom/sd-webui-depth-lib/issues/29
![image](https://github.com/jexom/sd-webui-depth-lib/assets/58225118/c4f80fd3-f201-4b58-89b6-a23abc2a3821)

**Comment by pipa0979** [comment]
> 2\. by another way to show the image lib, may could b

dataset = gr.Examples(examples=os.path.join(maps_path, t), inputs=[png_input_area],examples_per_page=24,label="Depth Maps", elem_id="examples")

Should be replaced by what?

**Comment by webcoderz** [comment]
gonna have to do this: https://github.com/jexom/sd-webui-depth-lib/issues/36#issue-1839673234

**Comment by jtydhr88** [comment]
@pipa0979 you don’t need to do that you pointed out, that’s some my personal suggestions for the extension author or someone who wants to contribute the fix. To fix it quickly you just need to do my first comment as @webcoderz showing

**Comment by ssrandur** [comment]
it works,thanks a lot

**Comment by Winters-Glade** [comment]
TLDR:
The problem for me was miaoshouai-assistant. Removed it, deleted venv folder. Works now.

Longer version:
I had tried everything I found on this page and a few others with no luck.
I can't say whether or not anything on this page helped my solution mentioned here work better or if there was no effect. All I  know is the fixes didn't work until I removed the miaoshouai-assistant extension. I really liked that extension and hope it's updated to working again for me.

Steps I performed to get my stable diffusion working again:

- Delete the venv folder
- Delete the Miaoshouai-assistant folder in /extensions 
- Run webui-user.bat

The venv folder was rebuilt and started working. The reason I take extra care to point this out is because I tried running SD with no extensions in the /extensions folder and still got the same error. I can only guess that miaoshouai-assistant puts something in the folder that causes the issue, or requires a dependency that causes the issue.

Hope this helps ♥

**Comment by XMUykyz** [comment]
> 你不需要这样做，你指出的，这是我对扩展作者或想要贡献修复的人的一些个人建议。要快速修复它，您只需要按照所示执行我的第一条评论即可

It worked for me. Thank you

**Comment by alexbofa** [comment]
@XMUykyz Hello, download this fork: https://github.com/wywywywy/sd-webui-depth-lib
